src/org/intellij/erlang/completion/ErlangAppCompletionContributor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2012-2015 Sergey Ignatov
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.intellij.erlang.completion;
+
+import com.intellij.codeInsight.completion.*;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.patterns.PatternCondition;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.ObjectUtils;
+import com.intellij.util.ProcessingContext;
+import com.intellij.util.containers.ContainerUtil;
+import org.intellij.erlang.ErlangFileType;
+import org.intellij.erlang.psi.*;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Set;
+
+import static com.intellij.patterns.PlatformPatterns.psiElement;
+import static com.intellij.patterns.PlatformPatterns.virtualFile;
+import static com.intellij.patterns.StandardPatterns.instanceOf;
+
+public class ErlangAppCompletionContributor extends CompletionContributor {
+  // source: "FILE SYNTAX" section of http://www.erlang.org/doc/man/app.html
+  public static final Set<String> KEYWORDS = ContainerUtil.newHashSet("description", "id", "vsn", "modules", "maxP",
+                                                                      "maxT", "registered", "included_applications",
+                                                                      "applications", "env", "mod", "start_phases",
+                                                                      "runtime_dependencies");
```

Comment:
It can't be public at least because it's mutable.

It's too bad it's a hash set in org.intellij.erlang.completion.ErlangKeywordsCompletionContributor - gotta get rid of that.

Concerning adding duplicate entries - it's a valid reason, indeed. Although, it should be a compile-time check which we cannot enforce, unfortunately.

All in all, there is no valid reason for storing these as hashes - it imposes extra overhead both in memory footprint and traversal time - we don't need that.